catalog/resource_sql_table.go

```diff
@@ -293,29 +293,55 @@ func (ti *SqlTableInfo) buildTableCreateStatement() string {
 	return strings.Join(statements, "")
 }
 
-// Wrapping the column name with backtiks to avoid special character messing things up.
+// Wrapping the column name with backticks to avoid special character messing things up.
 func getWrappedColumnName(ci SqlColumnInfo) string {
 	return fmt.Sprintf("`%s`", ci.Name)
 }
 
 func (ti *SqlTableInfo) getStatementsForColumnDiffs(oldti *SqlTableInfo, statements []string, typestring string) {
-	// TODO: take out "force_new" in `column` and add case to handle addition and removal of columns.
-	for i, ci := range ti.ColumnInfos {
-		oldCi := oldti.ColumnInfos[i]
-		if ci.Name != oldCi.Name {
-			statements = append(statements, fmt.Sprintf("ALTER %s %s RENAME COLUMN %s to %s", typestring, ti.SQLFullName(), getWrappedColumnName(oldCi), getWrappedColumnName(ci)))
+	if len(ti.ColumnInfos) != len(oldti.ColumnInfos) {
+		nameToOldColumn := make(map[string]SqlColumnInfo)
+		nameToNewColumn := make(map[string]SqlColumnInfo)
+		for _, ci := range oldti.ColumnInfos {
+			nameToOldColumn[ci.Name] = ci
 		}
-		if ci.Comment != oldCi.Comment {
-			statements = append(statements, fmt.Sprintf("ALTER %s %s ALTER COLUMN %s COMMENT %s", typestring, ti.SQLFullName(), getWrappedColumnName(ci), parseComment(ci.Comment)))
+		for _, newCi := range ti.ColumnInfos {
+			nameToNewColumn[newCi.Name] = newCi
 		}
-		if ci.Nullable != oldCi.Nullable {
-			var keyWord string
-			if ci.Nullable {
-				keyWord = "SET"
-			} else {
-				keyWord = "DROP"
+
+		for name, oldCi := range nameToOldColumn {
+			if _, exists := nameToNewColumn[name]; !exists {
+				// Remove old column if old column is no longer found in the config.
+				statements = append(statements, fmt.Sprintf("ALTER %s %s DROP COLUMN IF EXISTS %s", typestring, ti.SQLFullName(), getWrappedColumnName(oldCi)))
+			}
+		}
+
+		for name, newCi := range nameToNewColumn {
+			if _, exists := nameToOldColumn[name]; !exists {
+				// Add new column if new column is detected.
+				newCiStatement := ti.serializeColumnInfo(newCi)
+				statements = append(statements, fmt.Sprintf("ALTER %s %s ADD COLUMN %s ", typestring, ti.SQLFullName(), newCiStatement))
```

Comment:
it needs to be `ALTER TABLE table ADD COLUMN column data_type [COMMENT comment]`

Reply:
and we could batch multiple columns in a single alter statement
`ALTER TABLE table ADD COLUMNS (columnA data_type, columnB data_type)`